cmd/web3/main.go

```diff
@@ -678,11 +683,25 @@ func CallContract(ctx context.Context, rpcURL, privateKey, contractAddress, cont
 			if err != nil {
 				log.Fatalf("Cannot call the contract: %v", err)
 			}
-			receipt, err := web3.WaitForReceipt(ctx, client, tx.Hash)
-			if err != nil {
-				log.Fatalf("Cannot get the receipt: %v", err)
+			if waitForReceipt {
+				receipt, err := web3.WaitForReceipt(ctx, client, tx.Hash)
+				if err != nil {
+					log.Fatalf("Cannot get the receipt: %v", err)
+				}
+				logs, err := web3.ParseReceipt(myabi, receipt)
+				if err != nil {
+					log.Fatalf("Cannot parse the receipt logs: %v", err)
+				}
+				switch format {
+				case "json":
+					fmt.Println(marshalJSON(logs))
+					return
+				}
+				fmt.Println("Logs of the receipt", marshalJSON(logs))
+				fmt.Println("Transaction receipt address:", receipt.TxHash.Hex())
+			} else {
+				fmt.Println("Transaction address:", tx.Hash.Hex())
```

Comment:
This whole function is becoming difficult to follow. Can we refactor so that the short-circuit cases come first and return right away, and the long path stays unindented (no else)?